docs/html-templates.md

```diff
@@ -0,0 +1,40 @@
+# Html templates
+
+### Behavior
+
+* Templates are automatically recompiled in development when the file
+is saved; a refresh of the page should be enough to display the
+latest version.
+
+* The context, which sets the values that the variables in HTML
+templates should take, is defined in `zerver/context_processors.py`.
+Additional variable-setting might take place in the render call for
+that specific template.
+
+## Backend
+
+For text generated in the backend, including logged-out ("portico")
+pages and the webapp's base content, we use the [Jinja2][] template
+engine.
+
+The syntax for using conditionals and other common structures can be
+found [here][jconditionals].
+
+## Frontend
+
+For text generated in the frontend, live-rendering HTML from
+JavaScript for things like the main message feed, we use the
+[Handlebars][] template engine and sometimes work directly from
+JavaScript code.
+
+The syntax for using conditionals and other common structures can be
+found [here][hconditionals].
+
+[Jinja2]: http://jinja.pocoo.org/
```

Comment:
Probably worth adding a link back to translating along with a policy statement that all user-facing strings should be tagged for translation (with perhaps a sidenote that we don't consider pages only visible to sysadmins or developers to be user-facing for this purpose).

Also, we should search for links to translating.md and move the ones that should actually point here.